### PR TITLE
Added Scale to augment PrimaryMonitorSize.  Used gdi32 GetDeviceCaps …

### PR DESCRIPTION
**Pull request by jamessdixon**
Fixes #

## Proposed changes
Added Scale to Api
You don't have a DEV branch so I pushed to master

## Test methodology 
Integration tests

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/dotnet/winforms/pull/1486)

### CODE REVIEW

File: src/System.Windows.Forms/src/System/Windows/Forms/SystemInformation.cs [hunk 1]
```diff
@@ -71,6 +71,30 @@ public static Size PrimaryMonitorSize
                 UnsafeNativeMethods.GetSystemMetrics(NativeMethods.SM_CXSCREEN),
                 UnsafeNativeMethods.GetSystemMetrics(NativeMethods.SM_CYSCREEN));
 
+        [DllImport("gdi32.dll")]
+        static extern int GetDeviceCaps(IntPtr hdc, int nIndex);
```
Comment (hughbe) [inline src/System.Windows.Forms/src/System/Windows/Forms/SystemInformation.cs]: please use Interop.Gdi32.GetDeviceCaps

File: src/System.Windows.Forms/src/System/Windows/Forms/SystemInformation.cs [hunk 2]
```diff
@@ -71,6 +71,30 @@ public static Size PrimaryMonitorSize
...
+            DESKTOPVERTRES = 117,
+        }
+        /// <summary>
+        ///  Gets the scale of the primary display monitor.
+        /// </summary>
+        public static int Scale
+        {
+            get
+            {
+                var graphics = System.Drawing.Graphics.FromHwnd(IntPtr.Zero);
```
Comment (weltkante) [inline src/System.Windows.Forms/src/System/Windows/Forms/SystemInformation.cs]: Shouldn't this be in a using statement?

File: src/System.Windows.Forms/src/System/Windows/Forms/SystemInformation.cs [hunk 3]
```diff
@@ -71,6 +71,30 @@ public static Size PrimaryMonitorSize
                 UnsafeNativeMethods.GetSystemMetrics(NativeMethods.SM_CXSCREEN),
                 UnsafeNativeMethods.GetSystemMetrics(NativeMethods.SM_CYSCREEN));
 
+        [DllImport("gdi32.dll")]
+        static extern int GetDeviceCaps(IntPtr hdc, int nIndex);
+        public enum DeviceCap
```
Comment (weltkante) [inline src/System.Windows.Forms/src/System/Windows/Forms/SystemInformation.cs]: Why is this public?

File: src/System.Windows.Forms/src/System/Windows/Forms/SystemInformation.cs [hunk 4]
```diff
@@ -71,6 +71,30 @@ public static Size PrimaryMonitorSize
...
+        {
+            VERTRES = 10,
+            DESKTOPVERTRES = 117,
+        }
+        /// <summary>
+        ///  Gets the scale of the primary display monitor.
+        /// </summary>
+        public static int Scale
```
Comment (weltkante) [inline src/System.Windows.Forms/src/System/Windows/Forms/SystemInformation.cs]: This is new public API, not sure what else needs to be done (is a summary enough for documentation?)

**Comment by jamessdixon** [comment]
replaced Interop.Gdi32.GetDeviceCaps call

File: src/System.Windows.Forms/src/System/Windows/Forms/SystemInformation.cs [hunk 5]
```diff
@@ -71,6 +71,23 @@ public static Size PrimaryMonitorSize
...
 
+        /// <summary>
+        ///  Gets the scale of the primary display monitor.
+        /// </summary>
+        public static int Scale
+        {
+            get
+            {
+                var graphics = System.Drawing.Graphics.FromHwnd(IntPtr.Zero);
+                var desktop = graphics.GetHdc();
```
Comment (weltkante) [inline src/System.Windows.Forms/src/System/Windows/Forms/SystemInformation.cs]: [from documentation](https://docs.microsoft.com/en-us/dotnet/api/system.drawing.graphics.gethdc?view=netframework-4.8)
> Calls to the GetHdc and ReleaseHdc methods must appear in pairs.

You'll want a try-finally in addition to the using statement for the Graphics object itself.

**Comment by RussKie** [comment]
Thank you for the submission. 

We are not adding any new API for v3.0. 
We will consider new API for .NET Core 5.0, however all submissions will need to follow the established process (including the API Review). 

If you wish to pursue this further, please open a new issue and describe in details the problem and/or use-case you wish to address. A proposal on how the issue can be addressed is welcome too, as well as any considerations/implications associcated with the change.

**Pull request closed without merging**